src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -458,7 +458,7 @@ protected SSTableReader(Builder<?, ?> builder, Owner owner)
         this.openReason = builder.getOpenReason();
         this.first = builder.getFirst();
         this.last = builder.getLast();
-        this.bounds = AbstractBounds.strictlyWrapsAround(first.getToken(), last.getToken())
+        this.bounds = first == null || last == null || AbstractBounds.strictlyWrapsAround(first.getToken(), last.getToken())
```

Comment:
@jacek-lewandowski does this make sense to you? Ping me if you need more context

Reply:
ok, this is valid +1

Reply:
Just for future context: DTest scrub_test.TestScrub.test_standalone_scrub_essential_files_only exercises a path where we scrub sstables where some of the sstable files are missing. This can make those vars null so it's a rare but valid code path.